eng/New-TestResources.ps1

```diff
@@ -0,0 +1,305 @@
+
+#Requires -Version 6.0
+#Requires -PSEdition Core
+#Requires -Modules @{ModuleName='Az.Accounts'; ModuleVersion='1.6.4'}
+#Requires -Modules @{ModuleName='Az.Resources'; ModuleVersion='1.8.0'}
+
+[CmdletBinding(DefaultParameterSetName = 'Default', SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+param (
+    # Limit $BaseName to enough characters to be under limit plus prefixes, and https://docs.microsoft.com/azure/architecture/best-practices/resource-naming.
+    [Parameter(Mandatory = $true, Position = 0)]
+    [ValidatePattern('^[-a-zA-Z0-9\.\(\)_]{0,80}(?<=[a-zA-Z0-9\(\)])$')]
+    [string] $BaseName,
+
+    [Parameter(Mandatory = $true)]
+    [string] $ServiceDirectory,
+
+    [Parameter(Mandatory = $true)]
+    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+    [string] $TestApplicationId,
+
+    [Parameter()]
+    [string] $TestApplicationSecret,
+
+    [Parameter()]
+    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+    [string] $TestApplicationOid,
+
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $TenantId,
+
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+    [string] $ProvisionerApplicationId,
+
+    [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+    [string] $ProvisionerApplicationSecret,
+
+    [Parameter(ParameterSetName = 'Provisioner')]
+    [switch] $NoProvisionerAutoSave,
+
+    [Parameter()]
+    [ValidateRange(0, [int]::MaxValue)]
+    [int] $DeleteAfterHours,
+
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $Location = 'westus2',
+
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [hashtable] $AdditionalParameters,
+
+    [Parameter()]
+    [switch] $Force
+)
+
+# By default stop for any error.
+if (!$PSBoundParameters.ContainsKey('ErrorAction')) {
+    $ErrorActionPreference = 'Stop'
+}
+
+function Log($Message) {
+    Write-Host ('{0} - {1}' -f [DateTime]::Now.ToLongTimeString(), $Message)
+}
+
+# Support actions to invoke on exit.
+$exitActions = @({
+    if ($exitActions.Count -gt 1) {
+        Write-Verbose 'Running registered exit actions'
+    }
+})
+
+trap {
+    # Like using try..finally in PowerShell, but without keeping track of more braces or tabbing content.
+    $exitActions.Invoke()
+}
+
+# Enumerate test resources to deploy. Fail if none found.
+$root = Resolve-Path -Path "$PSScriptRoot/../sdk/$ServiceDirectory"
+$templateFileName = 'test-resources.json'
+$templateFiles = @()
+
+Write-Verbose "Checking for '$templateFileName' files under '$root'"
+Get-ChildItem -Path $root -Filter $templateFileName -Recurse | ForEach-Object {
+    $templateFile = $_.FullName
+
+    Write-Verbose "Found template '$templateFile'"
+    $templateFiles += $templateFile
+}
+
+if (!$templateFiles) {
+    Write-Warning -Message "No template files found under '$root'"
+    exit

```

Comment:
exit 1?

Reply:
That would be an error and, by default, fail the task. The idea here is that the script is always executed and will react accordingly.